client/src/test/scala/org/http4s/client/middleware/FollowRedirectSuite.scala

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s
+package client
+package middleware
+
+import cats.effect._
+import cats.effect.concurrent.Semaphore
+import cats.syntax.all._
+import java.util.concurrent.atomic._
+import org.http4s.Uri.uri
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.dsl.io._
+import org.http4s.syntax.all._
+import org.http4s.headers._
+
+class FollowRedirectSuite extends Http4sSuite with Http4sClientDsl[IO] {
+  private val loopCounter = new AtomicInteger(0)
+
+  val app = HttpRoutes
+    .of[IO] {
+      case GET -> Root / "loop" / i =>
+        val iteration = i.toInt
+        if (iteration < 3) {
+          val uri = Uri.unsafeFromString(s"/loop/${iteration + 1}")
+          MovedPermanently(Location(uri)).map(_.withEntity(iteration.toString))
+        } else
+          Ok(iteration.toString)
+
+      case req @ _ -> Root / "ok" =>
+        Ok(
+          req.body,
+          Header("X-Original-Method", req.method.toString),
+          Header(
+            "X-Original-Content-Length",
+            req.headers.get(`Content-Length`).fold(0L)(_.length).toString),
+          Header("X-Original-Authorization", req.headers.get(Authorization.name).fold("")(_.value))
+        )
+
+      case _ -> Root / "different-authority" =>
+        TemporaryRedirect(Location(uri("http://www.example.com/ok")))
+      case _ -> Root / status =>
+        Response[IO](status = Status.fromInt(status.toInt).yolo)
+          .putHeaders(Location(uri("/ok")))
+          .pure[IO]
+    }
+    .orNotFound
+
+  val defaultClient = Client.fromHttpApp(app)
+  val client = FollowRedirect(3)(defaultClient)
+
+  case class RedirectResponse(
+      method: String,
+      body: String
+  )
+
+  // test("FollowRedirect should ggfollow the proper strategy") {
```

Comment:
This test was marked as pending and it is a bit hard to port as it uses the `Tables` feature of spec. Instead I just commented it out. Perhaps it should be removed

Reply:
Drop it for now.  #3986 so it can be revived later, if we can.